packages/app/src/store/recoil/wallet.ts

```diff
@@ -83,14 +70,17 @@ export const latestNonceStore = selector({
   key: 'wallet.latest-nonce',
   get: ({ get }) => {
     const network = get(currentNetworkKeyStore);
-    const currentIdentity = get(currentIdentityStore);
-    const nonce = get(latestNoncesStore([network, currentIdentity?.getStxAddress() || '']));
+    const account = get(currentAccountStore);
+    const address = account
+      ? getStxAddress({ account, transactionVersion: TransactionVersion.Testnet })
+      : '';
+    const nonce = get(latestNoncesStore([network, address]));
     return nonce;
   },
 });
 
-export const currentIdentityIndexStore = atom<number | undefined>({
-  key: 'wallet.current-identity-index',
+export const currentAccountIndexStore = atom<number | undefined>({
+  key: 'wallet.current-account-index',
   default: undefined,
   effects_UNSTABLE: [localStorageEffect()],
```

Comment:
shouldn't this take a `key`?

Reply:
It's a builder-style function... `localStorageEffect()` returns a function that accepts the atom as it's first param (in line with the recoil effects API.